src/main/kotlin/com/hibob/academy/service/petService.kt

```diff
@@ -25,4 +25,20 @@ class PetService(private val petDao: PetDao) {
             throw BadRequestException("No pet with that id")
         }
     }
+
+    fun countPetsByType(companyId: Long): Map<String, Int> {
+        return petDao.countPetsByType(companyId)
+    }
+
+    fun adoptMultiple(pets: List<UUID>, ownerId: UUID, companyId: Long): Int {
+        var successfulyUpdated = 0
```

Comment:
We try to avoid using var

you can change you code to something like

pets.count { pet ->
    petDao.assignOwnerIdToPet(pet, ownerId) == 1
}
try and see if something like that works for you